### PR TITLE
Support external OTel exporters in CDI

### PR DESCRIPTION
**Pull request by brunobat**
When External exporters register an `ExternalOtelExporterBuildItem` the default exporter will not be instantiated. 
Includes small refactor and cleanup.

### CODE REVIEW

File: extensions/opentelemetry/deployment/src/main/java/io/quarkus/opentelemetry/deployment/OpenTelemetryBuildItem.java [hunk 1]
```diff
@@ -1,18 +0,0 @@
-package io.quarkus.opentelemetry.deployment;
-
-import io.opentelemetry.api.OpenTelemetry;
-import io.quarkus.builder.item.SimpleBuildItem;
-import io.quarkus.runtime.RuntimeValue;
-
-public final class OpenTelemetryBuildItem extends SimpleBuildItem {
```
Comment (geoand) [inline extensions/opentelemetry/deployment/src/main/java/io/quarkus/opentelemetry/deployment/OpenTelemetryBuildItem.java:7]: I assume this is no longer used by anyone?

Comment (brunobat) [reply extensions/opentelemetry/deployment/src/main/java/io/quarkus/opentelemetry/deployment/OpenTelemetryBuildItem.java:7]: No, it was dead code needed by the previous implementation of the exporter

**Review by geoand — APPROVED** [review]
LGTM

**Review by radcortez — COMMENTED** [review]
It makes sense to drop the default exporter if an extension registers its own, and if only one is supported.

In this case, it is possible to register multiple ones, so you can add multiple extensions and end up with multiple exporters. It may be acceptable to:

- I want to register this additional exporter and keep the default one
- I want to replace the default exporter

Are we sure we don't have a use case for the first one?

**Comment by brunobat** [comment]
> It makes sense to drop the default exporter if an extension registers its own, and if only one is supported.
> 
> In this case, it is possible to register multiple ones, so you can add multiple extensions and end up with multiple exporters. It may be acceptable to:
> 
> * I want to register this additional exporter and keep the default one
> * I want to replace the default exporter
> 
> Are we sure we don't have a use case for the first one?

It should be possible yes, but I haven't seen a request for it and because the Quarkiverse exporters are broken, I just wanted to get a quick fix out. 
I'll create an issue to track it.

**Pull request merged**